jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/learn/LearnCodeWhispererEditor.kt

```diff
@@ -0,0 +1,227 @@
+                        languageButtons.add(csharpButton)
+                        javaButton.component.doClick()
+                    }.bottomGap(BottomGap.MEDIUM)
+                }
+                row {
+                    cell(tryExamplePanel(project)).widthGroup(RIGHT_PANEL_WIDTH_GROUP)
+                }.bottomGap(BottomGap.MEDIUM)
+
+                // A separator with width adjusted to its sibling components
+                row {
+                    cell(SeparatorComponent(0, OnePixelDivider.BACKGROUND, null))
+                        .widthGroup(RIGHT_PANEL_WIDTH_GROUP)
+                }.bottomGap(BottomGap.MEDIUM)
+
+                title(message("codewhisperer.learn_page.codescan.title"))
+                row {
+                    cell(codeScanDescriptionPanel)
+                }.bottomGap(BottomGap.SMALL)
+                row {
+                    icon(ImageIcon(LearnCodeWhispererUIComponents.javaClass.classLoader.getResource("codewhisperer/codescan.png")))
+                        .widthGroup(RIGHT_PANEL_WIDTH_GROUP).align(AlignX.LEFT)
+                }
+            }
+        }
+    }
+    private val banner = panel {
+        panel {
+            customize(Gaps(10, 20, 10, 10))
+            row {
+                cell(bannerPanel()).resizableColumn().align(Align.FILL)
+            }
+        }
+    }.apply {
+        background = JBColor.BLUE.darker().darker()
+    }
+    private val rootPanel = panel {
+        val hasUserSeenNewUX = virtualFile.getUserData(NEW_ONBOARDING_UX_KEY) ?: false
+        if (!hasUserSeenNewUX) {
+            row {
+                cell(banner).resizableColumn().align(Align.FILL)
+            }
+        }
+        row {
+            scrollCell(contentPanel).align(Align.FILL)
+        }.resizableRow()
+    }
+
+    override fun getComponent(): JComponent = rootPanel
+
+    override fun getName(): String = "LearnCodeWhisperer"
+
+    override fun getPreferredFocusedComponent(): JComponent? = null
+
+    override fun isValid(): Boolean = true
+
+    override fun getCurrentLocation(): FileEditorLocation? = null
+
+    override fun getState(level: FileEditorStateLevel): FileEditorState = FileEditorState.INSTANCE
+
+    override fun isModified(): Boolean = false
+
+    override fun dispose() {}
+
+    override fun addPropertyChangeListener(listener: PropertyChangeListener) {}
+
+    override fun deselectNotify() {}
+
+    override fun getBackgroundHighlighter(): BackgroundEditorHighlighter? = null
+
+    override fun selectNotify() {}
+
+    override fun removePropertyChangeListener(listener: PropertyChangeListener) {}
+
+    override fun setState(state: FileEditorState) {}
+
+    override fun getFile(): VirtualFile = virtualFile
+
+    private fun Row.learnCodeWhispererLanguageButton(buttonLanguage: CodeWhispererProgrammingLanguage): Cell<JButton> {
+        val buttonContext = when (buttonLanguage) {
+            CodeWhispererJava.INSTANCE -> Tuple("Java ", AwsIcons.Misc.JAVA)
```

Comment:
The tuple could probably be replaced by a data class?

Reply:
yes -- if data class is preferred I will replace that, not a blocker.

Reply:
Yep, would prefer if it was changed, but can be a part of another pr!

Reply:
Just use the Kotlin built-in `Pair`

Reply:
I forgot that we can use Pair.